netket/driver/steady_state.py

```diff
@@ -121,7 +121,9 @@ def _forward_and_backward(self):
 
         # if it's the identity it does
         # self._dp = self._loss_grad
-        self._dp = self.preconditioner(self.state, self._loss_grad)
+        self._dp = self.preconditioner(
+            self.state, self._loss_grad, step_value=self.step_count
+        )
```

Comment:
this is not backward compatible: if I was using a custom preconditioner, this would break...

Reply:
I know, but there's no way around that, unless there's some way of checking what arguments a function accepts...

Reply:
Actually, we could put it in a try-except block and fall back on a call without `step_value`. I'm okay with that as long as we put a deprecation warning in the except block.

Reply:
This line is called once per driver iteration (**not** once per SR solver iteration, which would indeed be a problem). 

The only risk with the try-except approach is if someone uses a lazy custom-preconditioner constructor, which takes the extra arg, keeps it for a while and then errors out. I'm not too worried about this scenario as we're deprecating the whole alternative path anyway

Reply:
@PhilipVinc do you have any alternative suggestions or should we go ahead with the try-except solution?